src/m365/pp/commands/managementapp/managementapp-add.ts

```diff
@@ -0,0 +1,130 @@
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.appId = typeof args.options.appId !== 'undefined';
+    telemetryProps.objectId = typeof args.options.objectId !== 'undefined';
+    telemetryProps.name = typeof args.options.name !== 'undefined';
+    return telemetryProps;
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
+    this
+      .getAppObjectId(args)
+      .then((appId: string): Promise<void> => {
+        const requestOptions: any = {
+          // This should be refactored once we implement a PowerPlatform base class as api.bap will differ between envs.
+          url: `https://api.bap.microsoft.com/providers/Microsoft.BusinessAppPlatform/adminApplications/${appId}?api-version=2020-06-01`,
+          headers: {
+            accept: 'application/json;odata.metadata=none'
+          },
+          responseType: 'json'
+        };
+
+        return request.put(requestOptions);
+      })
+      .then((res: any): void => {
+        logger.log(res);
+        cb();
+      }, (err: any) => this.handleRejectedODataJsonPromise(err, logger, cb));
+  }
+
+  private getAppObjectId(args: CommandArgs): Promise<string> {
+    if (args.options.appId) {
+      return Promise.resolve(args.options.appId);
+    }
+
+    const { objectId, name } = args.options;
+
+    const filter: string = objectId ?
+      `id eq '${encodeURIComponent(objectId)}'` :
+      `displayName eq '${encodeURIComponent(name as string)}'`;
+
+    const requestOptions: any = {
+      url: `${this.resource}/v1.0/myorganization/applications?$filter=${filter}&$select=appId`,
+      headers: {
+        accept: 'application/json;odata.metadata=none'
+      },
+      responseType: 'json'
+    };
+
+    return request
+      .get<{ value: { appId: string }[] }>(requestOptions)
+      .then((res: { value: { appId: string }[] }): Promise<string> => {
+        if (res.value.length === 1) {
+          return Promise.resolve(res.value[0].appId);
+        }
+
+        if (res.value.length === 0) {
+          const applicationIdentifier = objectId ? `ID ${objectId}` : `name ${name}`;
+          return Promise.reject(`No Azure AD application registration with ${applicationIdentifier} found`);
+        }
+
+        return Promise.reject(`Multiple Azure AD application registration with name ${name} found. Please disambiguate (app IDs): ${res.value.map(a => a.appId).join(', ')}`);
+      });
+  }
+
+  public options(): CommandOption[] {
+    const options: CommandOption[] = [
+      { option: '--appId [appId]' },
+      { option: '--objectId [objectId]' },
+      { option: '--name [name]' }
+    ];
+
+    const parentOptions: CommandOption[] = super.options();
+    return options.concat(parentOptions);
+  }
+
+  public validate(args: CommandArgs): boolean | string {
+    if (!args.options.appId &&
```

Comment:
Shall we extend validation with checking if the access token is issued for the Power Platform app reg, like we discussed?

Reply:
@waldekmastykarz I can validate against `auth.service.appId` in the `validate` option. However once we have set CLI to allow this as well we can execute the command with the CLI ID, so the validation would fail but technically it would work. Any tips how we could tackle that one? Do we have something implemented in one the commands that I could use as inspiration?

Reply:
Just realize that we could do a check on the appId and add a `--confirm` flag if it is any other ID? Might that be an idea?

Reply:
Would it work with any other ID? I was under the impression that it must be the 1st part app or the request will fail, and if that's the case, then the `--confirm` option would have no use.

Reply:
@waldekmastykarz so that's where it get's weird. As soon as you have approved the CLI app Registration you can use that same app registration to approve others. Can't find any documentation on whether it is supported or not. But any App ID registered as a management then seems to be able to assign other management apps as well.  I am happy to implement any way we feel makes the most sense

Reply:
Gotcha. In that case, if that works, then we don't need to do any validation and can explain it in the docs.